src/main/java/org/sagebionetworks/web/client/SynapseStatusDetector.java

```diff
@@ -13,28 +20,52 @@
 	public static final String STATUS_PAGE_IO_PAGE = "kh896k90gyvg";
 	DateTimeFormat iso8601DateFormat = DateTimeFormat.getFormat(PredefinedFormat.ISO_8601);
 	DateTimeUtils dateTimeUtils;
-
+	StackConfigServiceAsync stackConfig;
+	SynapseJSNIUtils jsniUtils;
+	GlobalApplicationState globalAppState;
 	// how long the popup should be shown (in ms)
 	public static final int POPUP_DELAY_MS = 5000;
 
 	@Inject
-	public SynapseStatusDetector(GWTWrapper gwt, PopupUtilsView popupUtils, DateTimeUtils dateTimeUtils) {
+	public SynapseStatusDetector(GWTWrapper gwt, PopupUtilsView popupUtils, DateTimeUtils dateTimeUtils, StackConfigServiceAsync stackConfig, SynapseJSNIUtils jsniUtils, GlobalApplicationState globalAppState) {
 		this.gwt = gwt;
 		this.popupUtils = popupUtils;
 		this.dateTimeUtils = dateTimeUtils;
+		this.stackConfig = stackConfig;
+		this.jsniUtils = jsniUtils;
+		this.globalAppState = globalAppState;
 	}
 
 	public void start() {
 		_getCurrentStatus(this);
 		_getUnresolvedIncidents(this);
 		_getScheduledMaintenance(this);
-
+		getSynapseStackStatus();
+		
 		gwt.scheduleFixedDelay(() -> {
 			_getCurrentStatus(this);
 			_getUnresolvedIncidents(this);
+			getSynapseStackStatus();
```

Comment:
In addition to checking StatusPage for the status (and incidents), also check Synapse directly.  If we detect that we're in RO mode (or Down), redirect to the Down page (where it will click Back() when we're back in RW mode).